src/main/java/module/series/MatchDayPanel.java

```diff
@@ -117,13 +117,15 @@ private void setMatchButton(JButton button, Paarung paarung) {
         long gameFinishTime = 0;
         long nowTime = (new Date()).getTime();
 
+        boolean gameFinished = false;
         if (paarung != null) {
             gameFinishTime = paarung.getDatum().getTime();
             gameFinishTime = gameFinishTime + 3 * 60 * 60 * 1000L; //assuming 3 hours to make sure the game is finished
+            gameFinished = paarung.hatStattgefunden() == true || gameFinishTime < nowTime;
         }
 
         // Hat das Spiel schon stattgefunden
-        if ((paarung != null) && gameFinishTime < nowTime) {
+        if ((paarung != null) && gameFinished) {
```

Comment:
I don't see how this fix #337?  At first look the restriction seems even less restrictive than before ...